docs/self-managed/identity/user-guide/connect-to-an-existing-keycloak.md

```diff
@@ -0,0 +1,65 @@
+---
+id: connect-to-an-existing-keycloak
+title: "Connect to an existing Keycloak instance"
+sidebar_label: "Connect to an existing Keycloak instance"
+---
+
+In this guide, we'll demonstrate how to connect the Identity component to your existing Keycloak instance.
+
+### Prerequisites
+
+- Access to your Keycloak Admin Console
+- A basic understanding of administering realms and clients in Keycloak
+
+To connect Identity to an existing Keycloak instance, take the following steps:
+
+1. Log in to your Keycloak Admin Console.
+
+2. Hover over the selected realm name. In our example, this is **Master**. Select the realm you would like to connect Identity to. In our example, this is **Test Realm**.
+
+![keycloak-admin-realm-select](img/keycloak-admin-realm-select.png)
+
+3. Click **Clients** in the left navigation menu, and then click **Create**.
+
+![keycloak-admin-client-list](img/keycloak-admin-client-list.png)
+
+4. Enter the client ID and the URL of where your Identity instance will be hosted and click **Save**.
+   :::note What client ID should I use?
+   By default, Identity uses the Client ID `camunda-identity`, so we recommend using this too. If you choose a different client ID, this will need to be set in the Identity application [environment variables](/docs/self-managed/identity/deployment/configuration-variables.md).
+   :::
+
+![keycloak-admin-client-add](img/keycloak-admin-client-add.png)
+
+5. On the page for the client created, set the **Access Type** to `confidential` and **Service Accounts Enabled** to `ON`.
+
+![keycloak-admin-update-client-1](img/keycloak-admin-update-client-1.png)
+
+6. Scroll to the bottom of the page, click **Save**, and return to the top of the page.
+
+7. Click the **Service Account Roles** tab in the top navigation.
+
+![keycloak-admin-update-client-2](img/keycloak-admin-update-client-2.png)
+
+8. Select the `realm-management` client from the **Client Roles** dropdown.
+
+![keycloak-admin-update-client-3](img/keycloak-admin-update-client-3.png)
+
+9. Assign the `manage-clients`, `manage-realm`, and `manage-users` role from the **Available Roles** list.
+
+![keycloak-admin-update-client-4](img/keycloak-admin-update-client-4.png)
+
+:::note Why does Identity need these roles?
+Identity is designed to allow users to manage the various entities related to the Camunda Platform. To achieve this, it requires specific access to the realm.
+:::
+
+10. Click the **Credentials** tab and copy the client secret.
```

Comment:
Should they store this client secret somewhere?

Reply:
```suggestion
10. Navigate to the **Credentials** tab and copy the client secret.
```

Reply:
@christinaausley This step feels a little awkward but I couldn't really find the right way to word it, basically the client needs to take the value from here and then set the value in the Identity component configuration so that during startup Identity can connect successfully. 

This step and the next one are linked so the client should copy the secret from this page to their clipboard and then use it when configuring Identity to run.

Reply:
@Ben-Sheppard do you want me to take a look at this now? Or are you still making changes to the PR overall?

Reply:
A review would be appreciated (I'm not sure I have covered this comment well enough but I'm not sure we would want to say `copy to your clipboard`

Reply:
I think the way it's written and when you need to use the client secret is fast enough that this shouldn't be a problem. If we get feedback that it's confusing, let's change it in the future.